jsprocessor/SimpleWalletHandler.js

```diff
@@ -0,0 +1,213 @@
+  InternalError
+} = require('sawtooth-sdk/processor/exceptions')
+const crypto = require('crypto')
+const {TextEncoder, TextDecoder} = require('text-encoding/lib/encoding')
+
+const _hash = (x) => crypto.createHash('sha512').update(x).digest('hex').toLowerCase().substring(0, 64)
+var encoder = new TextEncoder('utf8')
+var decoder = new TextDecoder('utf8')
+const MIN_VALUE = 0
+const SW_FAMILY = 'simplewallet'
+const SW_NAMESPACE = _hash(SW_FAMILY).substring(0, 6)
+
+//function to obtain the payload obtained from the client
+const _decodeRequest = (payload) =>
+  new Promise((resolve, reject) => {
+    payload = payload.toString().split(',')
+    if (payload.length === 2) {
+      resolve({
+        action: payload[0],
+        amount: payload[1]
+      })
+    }
+   else if(payload.length === 3){ 
+	resolve({
+	  action:payload[0],
+	  amount:payload[1],
+	  toKey:payload[2]
+	})
+    }
+    else {
+      let reason = new InvalidTransaction('Invalid payload serialization')
+      reject(reason)
+    }
+})
+
+//function to display the errors
+const _toInternalError = (err) => {
+  console.log(" in error message block")
+  let message = err.message ? err.message : err
+  throw new InternalError(message)
+}
+
+//function to set the entries in the block using the "SetState" function
+const _setEntry = (context, address, stateValue) => {
+  let dataBytes = encoder.encode(stateValue)
+  let entries = {
+    [address]: dataBytes 
+  }
+  return context.setState(entries)
+}
+
+//function to make a deposit transaction
+const makeDeposit =(context, address, amount, user)  => (possibleAddressValues) => {
+  let stateValueRep = possibleAddressValues[address]
+  let newBalance = 0
+  let balance
+  if (stateValueRep == null || stateValueRep == ''){
+    console.log("No previous deposits, creating new deposit")
+    newBalance = amount
+  }
+  else{
+    balance = decoder.decode(stateValueRep)
+    newBalance = parseInt(balance) + amount
+    console.log("Amount crediting:"+newBalance)
+  }
+  let strNewBalance = newBalance.toString()
+  return _setEntry(context, address, strNewBalance)
+}
+
+//function to make a withdraw transaction
+const makeWithdraw =(context, address, amount, user)  => (possibleAddressValues) => {
+  let stateValueRep = possibleAddressValues[address]
+  let newBalance = 0
+  let balance
+  if (stateValueRep == null || stateValueRep == ''){
+    newBalance = amount
+  }
+  else{
+    balance = decoder.decode(stateValueRep)
+    balance = parseInt(balance)
+    if (balance < amount){
+      throw new InvalidTransaction(`Not enough money. The amount should be lesser or equal to ${balance}`)
+    }
+    newBalance = balance - amount
+    console.log("Amount debiting:"+newBalance)
+  }
+  let strNewBalance = newBalance.toString()
+  return _setEntry(context, address, strNewBalance)
+}
+
+//function to make a transfer transaction
+const makeTransfer =(context, senderAddress, amount, recieverAddress)  => (possibleAddressValues) => {
```

Comment:
Receiver and receiver are misspelled throughout this file:

s/eciever/eciever/g